### PR TITLE
New methods replace() and regexReplace()

### PR DESCRIPTION
**Pull request by asterixcapri**
I added docblocks and refactored place() to use replace() internally.

### CODE REVIEW

File: src/Task/File/Write.php [hunk 1]
```diff
@@ -31,33 +31,97 @@ public function __construct($filename)
...
+     *
+     * @param string $name
+     * @param string $val
+     *
+     * @return Write The current instance
+     */
     public function place($name, $val)
     {
-        $this->body = str_replace('{{' . $name . '}}', $val, $this->body);
+        $this->replace("{" . $name . "}", $val);
```
Comment (DavertMik) [inline src/Task/File/Write.php]: originally this was {{name}} (same as in handlebars)

**Comment by DavertMik** [comment]
Thanks, I'm merging this. I will restore {{ }} pattern in `place` by myself

**Comment by asterixcapri** [comment]
Thanks for merging. Regarding the braces, in revision bb57f68066580faba79ecf766ee1b6d9d36479f9 ```str_replace("{{$name}}", $val, $this->body)``` was replaced with ```str_replace("{{$name}}", $val, $this->body)```. This is wrong since the string "{{$name}}" is evaluated as "{value}".

**Comment by asterixcapri** [comment]
asterix@linuxdev:~$ php -r '$name = "value"; echo "{{$name}}\n";'
{value}

**Comment by boedah** [comment]
Yeah, that's why I wanted to clean up this statement with string concatenation (apart from PhpStorm choking on it...). 

Isn't there a test for that? 

My preference also is {{}}.
So this is a BC break.

**Comment by boedah** [comment]
Two other cool (tiny) features I would love:

* allow arbitrary amount of spaces around the variable name 
* allow placing an array of variables

**Pull request merged**